ext/opentelemetry-ext-grpc/src/opentelemetry/ext/grpc/_client.py

```diff
@@ -182,21 +227,36 @@ def intercept_stream(
             mutable_metadata = OrderedDict(metadata)
 
         with self._start_guarded_span(client_info.full_method) as guarded_span:
-            _inject_span_context(mutable_metadata)
-            metadata = tuple(mutable_metadata.items())
-            rpc_info = RpcInfo(
-                full_method=client_info.full_method,
-                metadata=metadata,
-                timeout=client_info.timeout,
-                request=request_or_iterator,
-            )
+            with self._metrics_recorder.record_latency(
+                client_info.full_method
+            ):
+                _inject_span_context(mutable_metadata)
+                metadata = tuple(mutable_metadata.items())
+                rpc_info = RpcInfo(
+                    full_method=client_info.full_method,
+                    metadata=metadata,
+                    timeout=client_info.timeout,
+                    request=request_or_iterator,
+                )
+
+                rpc_info.request = request_or_iterator
+
+                request_or_iterator = self._bytes_out_iterator_wrapper(
+                    request_or_iterator, client_info
+                )
+
+                try:
+                    result = invoker(request_or_iterator, metadata)
+                except grpc.RpcError as exc:
+                    guarded_span.generated_span.set_status(
+                        Status(StatusCanonicalCode(exc.code().value[0]))
+                    )
+                    raise
+
+                ret = self._trace_result(guarded_span, rpc_info, result)
```

Comment:
any reason for this return variable? it doesn't appear to be used anywhere other than in the return

Reply:
`_trace_result` assigned `rpc_info.response`, which is why I did it this way, but it does look pretty bad. Moved the byte count to inside `_trace_result` and got rid of this variable